task/build-vm-image/0.1/build-vm-image.yaml

```diff
@@ -226,8 +236,14 @@ spec:
           buildah --storage-driver=vfs manifest add --arch $(arch) --os linux --artifact --artifact-type application/vnd.diskimage.iso.gzip $OUTPUT_IMAGE /output/bootiso/install.iso.gz
         fi
         buildah --storage-driver=vfs manifest push --digestfile image-digest --authfile /.docker/config.json --all $OUTPUT_IMAGE
+
+        # At this point, we have pushed an image index of length 1 to the registry.
+        # Next, extract a reference to the image manifest and expose that, throwing away the image index.
+        IMAGE_INDEX_DIGEST=$(cat image-digest)
+        MANIFEST_DIGEST=$(buildah manifest inspect --authfile /.docker/config.json $OUTPUT_IMAGE@$IMAGE_INDEX_DIGEST | jq '.manifests[0].digest')
```

Comment:
It is a simpler change to continue to use buildah to push the artifacts. I first proposed the use of buildah so that we could generate the Image Index in the task as we were going to produce all artifacts together in a single task.

Since we have settled on this task to only produce a single disk image, does it make more sense to use a different tool that will _only_ push the image manifest and not the manifest + index? The closest approximation that buildah has is `buildah source` ... I have used that in the past but it doesn't allow for specifying the properties like artifacttype and platform.

Reply:
Yeah, I went down the same path thinking through the change. The usage of `oras` in the `oci-copy` task is probably pretty close to what we want to do here.

It'll be worth following up to see if that can make the implementation simpler.